Model/Service/Cache/CacheService.php

```diff
@@ -398,6 +398,20 @@ public function rebuildDirtyProduct(ProductCacheInterface $productIndex)
             );
             $store = $this->nostoHelperScope->getStore($productIndex->getStoreId());
             $nostoProduct = $this->nostoProductBuilder->build($magentoProduct, $store);
+            // We must remove the cache entry if product building returns null. This could happen if somebody
+            // for example uses the model filter or the building fails for some reason.
+            if ($nostoProduct === null) {
```

Comment:
This seems incorrect. Either the product is null due to an exception and we should catch that exception.